bodhi/tests/server/services/test_releases.py

```diff
@@ -251,6 +261,43 @@ def test_edit_release(self):
         r = self.db.query(Release).filter(Release.name == name).one()
         self.assertEquals(r.state, ReleaseState.current)
 
+    def test_edit_release_critpath_pkgs(self):
+        """
+        Assert that editing critical path packages for a release are saved correctly.
+        """
+        name = u"F22"
+
+        res = self.app.get('/releases/%s' % name, status=200)
+        r = res.json_body
+
+        r["edited"] = name
+        r["csrf_token"] = self.get_csrf_token()
+        r["critpath_pkgs"] = "bodhi"
+
+        res = self.app.post("/releases/", r, status=200)
+
+        r = self.db.query(Release).filter(Release.name == name).one()
+        self.assertEquals(r.critpath_pkgs[0].name, u'bodhi')
+        self.assertEquals(r.critpath_pkgs[0].critpath_releases[0].name, u'F22')
+
+    def test_edit_release_critpath_pkgs_duplicate_entry(self):
+        """
+        Assert that providing duplicate critical path packages for a release is handled correctly.
+        """
+        name = u"F22"
+
+        res = self.app.get('/releases/%s' % name, status=200)
+        r = res.json_body
+
+        r["edited"] = name
+        r["csrf_token"] = self.get_csrf_token()
+        r["critpath_pkgs"] = "bodhi,bodhi,bodhi"
+
+        res = self.app.post("/releases/", r, status=200)
+
+        r = self.db.query(Release).filter(Release.name == name).one()
+        self.assertEquals(len(r.critpath_pkgs), 1)
```

Comment:
We can use the stronger assertion I wrote above here to assert it is 1 package and that it is the correct one.